compileopts/config.go

```diff
@@ -238,6 +238,9 @@ func (c *Config) BinaryFormat(ext string) string {
 		// More information:
 		// https://github.com/Microsoft/uf2
 		return "uf2"
+	case ".zip":
+		// Nordic's proprietary DFU update package build by nrfutil
+		return "nrf-dfu"
```

Comment:
It is better to use nrf-dfu only when binary-format is specified for json.
The zip file format may be used for other purposes in the future.

You will need to change the JSON as well.


```suggestion
		if c.Target.BinaryFormat != "" {
			return c.Target.BinaryFormat
		}
		return "zip"
```

Reply:
I agree with @sago35 here.